src/legacy/core_plugins/interpreter/public/functions/esaggs.ts

```diff
@@ -112,13 +119,21 @@ export const esaggs = (): ExpressionFunction<typeof name, Context, Arguments, Re
       queryFilter,
     });
 
-    return {
+    const table: KibanaDatatable = {
       type: 'kibana_datatable',
       rows: response.rows,
-      columns: response.columns.map(column => ({
-        id: column.id,
-        name: column.name,
-      })),
+      columns: response.columns.map((column: any) => {
+        const cleanedColumn: KibanaDatatable['columns'][0] = {
```

Comment:
Do you think enough folks will have a use case for this that it'd be worth exporting `Column` separately, perhaps as `KibanaDatatableColumn`, so that you don't have to do stuff like this?

Reply:
why do we always take the first column ?

Reply:
Because all of the columns in the array have the same type, this is a hack. I agree with @lukeelmers here, it makes sense to expose `KibanaDatatableColumn` to be able to cleanly reference this type.

Current solution works like this:
`KibanaDatatable` is of type `{ columns: Array<Column> }`
`KibanaDatatable['columns]` is of type `Array<Column>`

now the question is how do I get to the type of the Column itself? I could do a proper type inference like this: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html#type-inference-in-conditional-types or I could do a dirty hack and get the type of the `0` property of `KibanaDatatable['columns]` which happens to be `Column`

Reply:
A bit orthogonal to the discussion here, but in cases where magic like this is necessary, I find the syntax `KibanaDatatable['columns'][number]` to be slightly easier to grok, mostly because the index number can be distracting as @ppisljar points out.

But agreed that exporting separately feels like the way to go here.